owncloudApp/src/main/java/com/owncloud/android/extensions/ActivityExt.kt

```diff
@@ -302,3 +330,7 @@ fun Activity.showBiometricDialog(iEnableBiometrics: IEnableBiometrics) {
     }.show()
 }
 
+enum class LockEnforcedType {
+    DISABLED, DIALOG_OPTION, PASSCODE_OPTION, PATTERN_OPTION
+}
```

Comment:
I agree with you that it can be moved to an isolated file, but I am not very much in favor of parsing using a where condition, I think it is more explanatory this way, what do you think?

```
enum class LockEnforcedType(val value: Int) {
    DISABLED(0), EITHER_ENFORCED(1), PASSCODE_ENFORCED(2), PATTERN_ENFORCED(3);

    companion object {
        fun toLockEnforcedType(value: Int): LockEnforcedType = values().first { it.value == value }
    }
}
```

Reply:
I consider that it is not more explanatory, it is by far more confusing that way from my point of view.

When using it:
`     val lockEnforcedType = LockEnforcedType.toLockEnforcedType(2)
`
vs
`    val lockEnforcedType = LockEnforcedType.parseFromInteger(2)
`

About the `when` clause, I think it is easier to know which is the default value and what to do when receiving a non-expected value. If we receive a 4 (we won't have restricted values via MDM), do we expect to throw an exception or make it disabled by default?

Reply:
> I consider that it is not more explanatory

Maybe the name is not the most appropriate, but I still think that having the values defined (no other value is expected to arrive) I prefer to use a line of code to see if that value exists within the enum and not when clause. Everyone to his own taste. 😉 

> we won't have restricted values via MDM

I was not aware of this, in this case it is correct to use when clause.